src/cmd/flux-kvs.c

```diff
@@ -816,6 +851,254 @@ int cmd_dir (optparse_t *p, int argc, char **argv)
     return (0);
 }
 
+/* Find longest name in a directory.
+ */
+static int get_dir_maxname (kvsdir_t *dir)
+{
+    kvsitr_t *itr;
+    const char *name;
+    int max = 0;
+
+    if (!(itr = kvsitr_create (dir)))
+        log_err_exit ("kvsitr_create");
+    while ((name = kvsitr_next (itr)))
+        if (max < strlen (name))
+            max = strlen (name);
+    kvsitr_destroy (itr);
+    return max;
+}
+
+/* Find longest name in a list of names.
+ */
+static int get_list_maxname (zlist_t *names)
+{
+    const char *name;
+    int max = 0;
+
+    name = zlist_first (names);
+    while (name) {
+        if (max < strlen (name))
+            max = strlen (name);
+        name = zlist_next (names);
+    }
+    return max;
+}
+
+/* Determine appropriate terminal window width for columnar output.
+ * The options -w COLS or -1 force the width to COLS or 1, respectively.
+ * If not forced, ask the tty how wide it is.  If not on a tty, assume 80
+ * if -C, or 1 if not.
+ * N.B. A width of 0 means "unlimited", while a width of 1 effectively
+ * forces one entry per line since entries are never broken across lines.
+ */
```

Comment:
To match `ls` should `get_window_width()` also check `COLUMNS` env var? From GNU `ls(1)`:
``` 
‘-w cols’
‘--width=cols’
Assume the screen is cols columns wide. The default is taken from the terminal settings if possible; otherwise the environment variable COLUMNS is used if it is set; otherwise the default is 80. With a cols value of ‘0’, there is no limit on the length of the output line, and that single output line will be delimited with spaces, not tabs.
```

Reply:
Good catch, I did not read that carefully in the ls(1) page.  Will fix.